identity-service/src/routes/tiktok.js

```diff
@@ -57,9 +64,159 @@ module.exports = function (app) {
       urlAccessToken += '&code=' + code
       urlAccessToken += '&grant_type=authorization_code'
 
-      const resp = await axios.post(urlAccessToken)
+      try {
+        // Fetch user's accessToken
+        const accessTokenResponse = await axios.post(urlAccessToken)
+        const { access_token: accessToken } = accessTokenResponse.data
+
+        // Fetch TikTok user from the TikTok API
+        const userResponse = await axios.post(
+          `https://open-api.tiktok.com/user/info/?access_token=${accessToken}`,
+          {
+            fields: [
+              'open_id',
+              'display_name',
+              'profile_deep_link',
+              'is_verified'
+            ]
+          }
+        )
+
+        const { data, error } = userResponse.data
+
+        if (error.code) {
+          return errorResponseBadRequest(error.message)
+        }
+
+        const { user: tikTokUser } = data
+
+        const existingTikTokUser = await models.TikTokUser.findOne({
+          where: {
+            uuid: tikTokUser.open_id,
+            blockchainUserId: {
+              [models.Sequelize.Op.not]: null
+            }
+          }
+        })
+
+        if (existingTikTokUser) {
```

Comment:
All good for logic, but guessing we could do the upset in the if statement, and then unconditionally return the data? And what about the error case? Do we still want to return the data? If not we could prob just rely out outer catch?

Reply:
Agreed, I'll update this. In the case where the request for the user fails we still want to return the access token